File: source/downloads/code/LeetCode/Java/Pascal-Triangle.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ret = new ArrayList<List<Integer>>();
        if (numRows<=0)
            return ret;
        for (int i=0;i<numRows;i++){
            List<Integer> row = new ArrayList<Integer>();
            for(int j=0;j<i+1;j++){
                if (j==0 || j==i)
                    row.add(1);
                else
                    row.add(ret.get(i-1).get(j-1) + ret.get(i-1).get(j));
            }
            ret.add(row);
        }       
       return ret; 
    }
}

File: downloads/code/LeetCode/Python/Pascal-Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        ret = []
        if numRows < 1:
            return ret
        ret.append([1])
        if numRows == 1:
            return ret

        for i in range(1,numRows):
            tmp = [1]
            for j in range(1, i):
                tmp.append(ret[i-1][j] + ret[i-1][j-1])
            tmp.append(1)
            ret.append(tmp[:])
        return ret
